hibernate-core/src/main/java/org/hibernate/type/InstantType.java

```diff
@@ -32,15 +33,16 @@
 	 */
 	public static final InstantType INSTANCE = new InstantType();
 
-	public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern( "yyyy-MM-dd HH:mm:ss.S 'Z'", Locale.ENGLISH );
+	public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern( "yyyy-MM-dd HH:mm:ss.n ZZZZ", Locale.ENGLISH );
```

Comment:
Why did you change from `S` (fractions of a second) to `n` (nanoseconds)?

Why `ZZZZ` (the long TimeZone format) instead of `Z`?